helix-term/src/ui/mod.rs

```diff
@@ -225,20 +209,27 @@ pub fn file_picker(root: PathBuf, config: &helix_view::editor::Config) -> Picker
             cx.editor.set_error(err);
         }
     })
-    .with_preview(|_editor, path| Some((path.clone().into(), None)))
+    .with_preview(|_editor, path| Some((path.clone().into(), None)));
+    let injector = picker.injector();
+    std::thread::spawn(move || {
+        for file in files {
+            if injector.push(file).is_err() {
+                break;
+            }
+        }
+    });
```

Comment:
I've seen this in a couple places, where we're spinning up a dedicated thread for these operations. Why is this preferable to a tokio task? Is this just an effort to avoid interfering with the main event loop?

Are there any downsides to this approach? If I'm understanding this right, this is basically fire and forget, and if the thread errors or panics, there's nothing we can do, is that right?

Reply:
You are not supposed to block in async takes so tokio::spawn would be wrong to use here (it messes up tokio scheduling). We could use the blocking task spawner but in the best case that will reuse an existing sleeping thread (but in moat cases it would also just spawn a new thread itself). I could move to that but it ultimately doesn't really make a difference so I just reached for the standard library.

Ultimately for heavy compute and blocking io (you cant make readdir async the kernel doesnt support that) on a background thread that is throughput bound instead of waiting for a message or something tokio doesn't really offer any advantages (in fact the docs even recommknd rayon for that).

I guess it's true that we would miss panics here but I don't consider that a huge issue. A panic here means a panic in walkdir/ignore (it doesn't do much else). In the worst case you will stop seeing new files stream into the matcher.

Reply:
Makes sense, thanks for the explanation